templates/user_form2.tt.html

```diff
@@ -0,0 +1,241 @@
+    value = value.replace(new RegExp("[^a-zA-Z0-9-]", 'g'),"-");
+    value = value.replace(new RegExp("-+", 'g'),"-");
+    value = value.replace(new RegExp("^-"),"");
+
+    inputfield.value = value;
+    }
+</script>
+
+[% IF action == 'display' || action == 'none' %]
+[% INCLUDE 'error_list.tt.html' %]
+[% END %]
+
+[% IF action == 'display' %]
+
+    <!-- Start form -->
+
+    <form method="post" action="/cgi/user2.pl" enctype="multipart/form-data">
+		[% FOREACH section IN sections %]
+			[% IF user OR section.id == "user" %]
+				[% FOREACH field IN section.fields %]
+					[% IF field.field == 'userid' %]
+                        [% IF type == 'edit' %]
+						    <label for="[% field.field %]"> [% lang(field.label2) %]</label>
+                            <p> [% user_id_field %] </p>
+                        [% ELSE %]
+                            <label for="[% field.field %]">  [% lang(field.label2) %] [% lang('username_info')  %]</label>
+                            <input type="text" id="[% field.field %]" name="[% field.field %]" value="[% field.value %]" autocomplete="[% field.field %]" onkeyup="normalize_string_value(this)" />
+                        [% END %]
+					[% ELSIF field.type == 'text' %]
+                        <label for="[% field.field %]">[% field.label %]</label>
+                        <input type="text" id="[% field.field %]" name="[% field.field %]" value="[% field.value %]" autocomplete="[% field.field %]"/>
+					[% ELSIF field.type == 'password' %]
+						<label for="[% field.field %]">[% lang(field.label2) %]</label>
+						<input type="password" id="[% field.field %]" name="[% field.field %]" value="[% user_password_field %]" autocomplete="new-password"/>
+					[% ELSIF field.type == 'textarea' %]
+						<label for="[% field.field %]">[% field.label %]</label>
+						<textarea id="[% field.field %]" name="[% field.field %]" style="height:100px;" autocomplete="[% field.field %]">[% field.value %]</textarea>
+					[% ELSIF field.type == 'email' %]
+						<label for="[% field.field %]">	[% field.label %]</label>
+						<input type="email" id="[% field.field %]" name="[% field.field %]" value = "[% field.value %]" autocomplete="[% field.field %]"/>
+					[% END %]
+				[% END %]
+			[% END %]
+
+			[% IF professional OR section.id == "professional" %]
+
+                <h5>[% lang("pro_account") %]</h5>
+
+                <p>[% lang("if_you_work_for_a_producer") %]</p>
+                <p>[% lang("producers_platform_description_long") %]</p>
+
+                [% IF accepted_organization.defined %]
+                    <p>[% pro_account_org %]</p>
+                [% ELSIF product_type.defined && product_type == "food" %]
+                    [% FOREACH field IN section.fields %]
+                        [% IF field.type == 'checkbox' %]
+                            <input type="checkbox" id="[% field.field %]" name="[% field.field %]" [% IF pro_checked == 'on' %]checked="checked"[% END %] />
```

Comment:
Here you could do it like in the org template:

						<label for="[% field.field %]">
							<input type="checkbox" id="[% field.field %]" name="[% field.field %]" [% IF field.value == 'on' %]checked="checked"[% END %] />
							[% field.label %]
						</label>

Reply:
The issue here was that in `users.pm` it is used like this. The label is associated with a hidden field. I first tried it like this, but it was causing issues of different names I guess.  So I tried to stick to the `Users.pm` to avoid errors.

Here is the code snippet of `Users.pm`:

`"<input type=\"checkbox\" id=\"pro\" name=\"pro\" $pro_checked>"
		. "<label for=\"pro\">"
		. "<input type=\"hidden\" name=\"pro_checkbox\" value=\"1\"> "
		. lang("this_is_a_pro_account") . "</label>"`